### PR TITLE
[cmake] set correct ARCH for iOS

### PR DESCRIPTION
**Pull request by Rechi**
## Types of change
- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

**Review by Memphiz — COMMENTED** [review]
What exactly does this influence? Does this change some addon related binary names aswell? The former arm-osx sounds familiar to me but I can‘t remember where I saw this somewhere in a dylib filename.

**Comment by Rechi** [comment]
You already asked at https://github.com/xbmc/xbmc/pull/9806#discussion_r63074534 why adding this hack, but sadly you didn't insist on not adding it.

#14250 showed me that something is wrong with iOS as aarch64 was detected as arm.

File: cmake/scripts/ios/ArchSetup.cmake [hunk 1]
```diff
@@ -12,9 +12,13 @@ set(CMAKE_SYSTEM_NAME Darwin)
...
-  if(CPU STREQUAL armv7 OR CPU STREQUAL arm64)
+  if(CPU STREQUAL armv7)
     set(CMAKE_OSX_ARCHITECTURES ${CPU})
-    set(ARCH arm-osx)
+    set(ARCH arm)
+    set(NEON True)
+  elseif(CPU STREQUAL arm64)
+    set(CMAKE_OSX_ARCHITECTURES ${CPU})
+    set(ARCH aarch64)
     set(NEON True)
```
Comment (Memphiz) [inline cmake/scripts/ios/ArchSetup.cmake]: This one is still correct then?

**Review by Memphiz — APPROVED** [review]
I am pretty sure that arm-osx was used in some header for generating dylib names at the time I commented on the other PR. Seems there is no trace of this left beside in my brain ;) - so go for it

File: cmake/scripts/ios/ArchSetup.cmake (hunk 1 above)
Comment (Rechi) [reply cmake/scripts/ios/ArchSetup.cmake]: It's the same as before.

**Pull request merged**